docs/JSON_format.md

```diff
@@ -419,12 +425,44 @@ attributes for these objects are:
     (#the-type-value-object) for more information on expressions format.
 
 The `match_type` for the table needs to follow the following rules:
+- If one match field is `range`, the table `match_type` has to be `range`
 - If one match field is `ternary`, the table `match_type` has to be `ternary`
 - If one match field is `lpm`, the table `match_type` is either `ternary` or
 `lpm`
 Note that it is not correct to have more than one `lpm` match field in the same
 table.
 
+#### match-action entry format
+
+We describe the format of the match-action entries contained in the `entries`
+JSON array (see table JSON format above). Each entry is a JSON object with the
+following attributes:
+- `match_key`: a JSON array of objects with the following attributes:
+  - `match_type`: one of `valid`, `exact`, `lpm`, `ternary`, `range`
+  - the other attributes depend on the `match_type`:
+    - for `valid`: we need a `key` attribute which has to be a boolean
+    - for `exact`: we need a `key` attribute which has to be a hexstring
+    - for `lpm`: we need a `key` attribute which has to be a hexstring and a
+    `prefix_length` attribute which has to be an integer
+    - for `ternary`: we need a `key` and a `mask` attributes which both have to
+    be hexstrings
+    - for `range`: we need a `start` and a `end` attribute which both have to be
+    hexstrings
+- `action_entry`: a JSON object with the following attributes:
+  - `action_id`: the id of the default action
+  - `action_data`: a JSON array where each entry is the hexstring value for an
+  action argument. The size of the array needs to match the number of parameters
+  expected by the action function with id `action_id`.
```

Comment:
I think that regardless of what we decide on the language side, the compiler should assign a priority to the entry.